src/diagonal.works/b6/graph/graph.go

```diff
@@ -149,6 +149,42 @@ func IsPathUsableByPedestrian(path b6.PathFeature) bool {
 	return false
 }
 
+type ElevationWeights struct{}
+
+func (ElevationWeights) IsUseable(segment b6.Segment) bool {
```

Comment:
When you come to integrate this, maybe you want to delegate the IsUseable call to another Weights instance (for example, SimpleHighwayWeights) to take into account the type of the path too without duplicating logic. Right now you'll have people traversing the edge of buildings, walls etc. In which case, you could consider Weight() as multiplying the value from the underlying Weights instance, rather than calculating the length here.